src/vs/workbench/contrib/tasks/browser/taskTerminalStatus.ts

```diff
@@ -61,16 +68,45 @@ export class TaskTerminalStatus extends Disposable {
 	addTerminal(task: Task, terminal: ITerminalInstance, problemMatcher: AbstractProblemCollector) {
 		const status: ITerminalStatus = { id: TASK_TERMINAL_STATUS_ID, severity: Severity.Info };
 		terminal.statusList.add(status);
+
+		this.terminalMarkerMap.set(terminal.instanceId, {
+			resources: new Map<string, URI>(),
+			markers: new Map<string, Map<string, IMarkerData>>()
+		});
+		this._register(terminal.onDisposed(() => {
+			this.terminalMarkerMap.delete(terminal.instanceId);
+		}));
 		this._register(problemMatcher.onDidFindFirstMatch(() => {
 			this._marker = terminal.registerMarker();
 			if (this._marker) {
 				this._register(this._marker);
 			}
 		}));
-		this._register(problemMatcher.onDidFindErrors(() => {
+		this._register(problemMatcher.onDidFindErrors((markers: ITaskMarker[]) => {
```

Comment:
Could we move this to `chatAgentTools/browser/taskProblemMonitor.ts`?

Reply:
we don't have access to the `problemMatcher` from the `tasksService`, so would need to plumb that through anyway